src/Interpreters/ClientInfo.cpp

```diff
@@ -113,6 +132,26 @@ void ClientInfo::read(ReadBuffer & in, const UInt64 client_protocol_revision)
         else
             client_version_patch = client_revision;
     }
+
+    // TODO what does it even mean to read this structure over HTTP? I thought
+    // this was for native protocol? See interface == Interface::HTTP.
+    if (client_protocol_revision >= DBMS_MIN_REVISION_WITH_OPENTELEMETRY)
+    {
+        uint8_t have_trace_id = 0;
+        readBinary(have_trace_id, in);
+        if (have_trace_id)
+        {
+            readBinary(opentelemetry_trace_id, in);
+            readBinary(opentelemetry_span_id, in);
+            readBinary(opentelemetry_parent_span_id, in);
+            readBinary(opentelemetry_tracestate, in);
+            readBinary(opentelemetry_trace_flags, in);
+
+            fmt::print(stderr, "read {:x}, {}, {} at\n{}\n",
```

Comment:
This looks like it shouldn't be here.

Reply:
Yep, this draft still has some debug output here and there, will remove.